editor/src/components/canvas/canvas-strategies/strategies/reparent-utils.ts

```diff
@@ -87,6 +88,43 @@ export function elementToReparent(element: JSXElementChild, imports: Imports): E
 
 export type ToReparent = PathToReparent | ElementToReparent
 
+function adjustElementDuplicateName(
+  element: ElementToReparent,
+  targetFile: string,
+  projectContents: ProjectContentTreeRoot,
+): ElementToReparent {
+  const fileContents = getParseSuccessForFilePath(targetFile, projectContents)
+  if (fileContents == null) {
+    return element
+  }
+  const { imports, duplicateNameMapping } = renameDuplicateImports(
+    fileContents.imports,
+    element.imports,
+    targetFile,
+  )
+  // handle element name
+  if (isJSXElement(element.element)) {
+    const elementName = MetadataUtils.getJSXElementName(element.element)?.baseVariable
+    if (elementName != null && duplicateNameMapping.has(elementName)) {
+      element.element.name.baseVariable = duplicateNameMapping.get(elementName)!
```

Comment:
Generally we try to avoid mutating incoming parameters like this, it would be better to create a new value and return that so as not to subvert that expectation.